source/sigv4.c

```diff
@@ -1162,110 +1621,1311 @@ static SigV4Status_t getCredentialScope( SigV4Parameters_t * pSigV4Params,
+                                        size_t algorithmLen,
+                                        CanonicalContext_t * pCanonicalContext )
+{
+    SigV4Status_t returnStatus = SigV4Success;
+    size_t encodedLen = pCanonicalContext->bufRemaining;
+    char * pBufStart = ( char * ) pCanonicalContext->pBufProcessing;
+
+    assert( pParams != NULL );
+    assert( ( pAlgorithm != NULL ) && ( algorithmLen > 0 ) );
+    assert( pCanonicalContext != NULL );
+
+    returnStatus = completeHashAndHexEncode( pBufStart,
+                                             ( size_t ) ( pCanonicalContext->pBufCur - pBufStart ),
+                                             pCanonicalContext->pBufCur + 1,
+                                             &encodedLen,
+                                             pParams->pCryptoInterface );
+
+    if( returnStatus == SigV4Success )
+    {
+        size_t sizeNeededBeforeHash = algorithmLen + 1U +         \
+                                      SIGV4_ISO_STRING_LEN + 1U + \
+                                      sizeNeededForCredentialScope( pParams ) + 1U;
+
+        /* Check if there is enough space for the string to sign. */
+        if( sizeNeededBeforeHash + ( pParams->pCryptoInterface->hashDigestLen * 2U ) >
+            SIGV4_PROCESSING_BUFFER_LENGTH )
+        {
+            returnStatus = SigV4InsufficientMemory;
+            LOG_INSUFFICIENT_MEMORY_ERROR( "for string to sign",
+                                           sizeNeededBeforeHash + ( pParams->pCryptoInterface->hashDigestLen * 2U ) - SIGV4_PROCESSING_BUFFER_LENGTH );
+        }
+        else
+        {
+            /* Copy the hash of the canonical request beforehand to its precalculated location
+             * in the string to sign. */
+            ( void ) memmove( pBufStart + sizeNeededBeforeHash,
+                              pCanonicalContext->pBufCur + 1,
+                              encodedLen );
+            pCanonicalContext->pBufCur = pBufStart + sizeNeededBeforeHash + encodedLen;
+            pCanonicalContext->bufRemaining = SIGV4_PROCESSING_BUFFER_LENGTH - encodedLen - sizeNeededBeforeHash;
+        }
+    }
+
+    if( returnStatus == SigV4Success )
+    {
+        size_t bytesWritten = 0U;
+        SigV4String_t credentialScope;
+        bytesWritten = writeStringToSignPrefix( pBufStart,
+                                                pAlgorithm,
+                                                algorithmLen,
+                                                pParams->pDateIso8601 );
+        pBufStart += bytesWritten;
+        credentialScope.pData = pBufStart;
+        credentialScope.dataLen = sizeNeededForCredentialScope( pParams );
+        /* Concatenate credential scope. */
+        ( void ) generateCredentialScope( pParams, &credentialScope );
+        pBufStart += credentialScope.dataLen;
+        /* Concatenate linefeed character. */
+        *pBufStart = LINEFEED_CHAR;
+    }
+
+    return returnStatus;
+}
+
```

Comment:
Same as above Functions needs to be separated using `/*-----------------------------------------------------------*/`